solang-parser/src/helpers/fmt.rs

```diff
@@ -0,0 +1,2501 @@
+                Ok(())
+            }
+            Self::NumberLiteral(_, value, exponent, ident) => {
+                f.write_str(value)?;
+                if !exponent.is_empty() {
+                    f.write_char('e')?;
+                    f.write_str(exponent)?;
+                }
+                write_opt!(f, ": ", ident);
+                Ok(())
+            }
+            Self::HexNumberLiteral(_, value, ident) => {
+                f.write_str(value)?;
+                write_opt!(f, ": ", ident);
+                Ok(())
+            }
+            Self::HexStringLiteral(value, ident) => {
+                value.fmt(f)?;
+                write_opt!(f, ": ", ident);
+                Ok(())
+            }
+            Self::StringLiteral(value, ident) => {
+                value.fmt(f)?;
+                write_opt!(f, ": ", ident);
+                Ok(())
+            }
+            Self::Variable(ident) => ident.fmt(f),
+            Self::FunctionCall(call) => call.fmt(f),
+            Self::SuffixAccess(_, l, r) => {
+                l.fmt(f)?;
+                f.write_char('.')?;
+                r.fmt(f)
+            }
+        }
+    }
+}
+
+impl Display for pt::YulStatement {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        match self {
+            Self::Block(inner) => inner.fmt(f),
+            Self::FunctionDefinition(inner) => inner.fmt(f),
+            Self::FunctionCall(inner) => inner.fmt(f),
+            Self::For(inner) => inner.fmt(f),
+            Self::Switch(inner) => inner.fmt(f),
+
+            Self::Assign(_, exprs, eq_expr) => {
+                write_separated(exprs, f, ", ")?;
+                f.write_str(" := ")?;
+                eq_expr.fmt(f)
+            }
+            Self::VariableDeclaration(_, vars, eq_expr) => {
+                f.write_str("let")?;
+                if !vars.is_empty() {
+                    f.write_char(' ')?;
+                    write_separated(vars, f, ", ")?;
+                }
+                write_opt!(f, " := ", eq_expr);
+                Ok(())
+            }
+
+            Self::If(_, expr, block) => {
+                f.write_str("if ")?;
+                expr.fmt(f)?;
+                f.write_char(' ')?;
+                block.fmt(f)
+            }
+
+            Self::Leave(_) => f.write_str("leave"),
+            Self::Break(_) => f.write_str("break"),
+            Self::Continue(_) => f.write_str("continue"),
+
+            Self::Error(_) => Ok(()),
+        }
+    }
+}
+
+impl Display for pt::YulSwitchOptions {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        match self {
+            Self::Case(_, expr, block) => {
+                f.write_str("case ")?;
+                expr.fmt(f)?;
+                f.write_str(" ")?;
+                block.fmt(f)
+            }
+            Self::Default(_, block) => {
+                f.write_str("default ")?;
+                block.fmt(f)
+            }
+        }
+    }
+}
+
+#[inline]
```

Comment:
It is obvious why it should be inlined. What is not obvious is why the compiler doesn't do it automatically or why we think it needs this hint (we _don't_ know) and so why the need for this attribute arises.

I left the comments as a suggestion, WDYT? :upside_down_face: 

```suggestion
// This function is private so it should be inlined by the compiler anyways. 
// We provided this as an extra hint regardless because we don't expect compile time penalties or other negative impacts from it.
#[inline]
```